blaze/compute/sql.py

```diff
@@ -585,6 +593,24 @@ def compute_up(expr, data, **kwargs):
     return table_of_engine(data, expr._name)
 
 
+@dispatch(DateTime, (ClauseElement, sa.sql.elements.ColumnElement))
+def compute_up(expr, data, **kwargs):
+    return sa.extract(expr.attr, data).label(expr._name)
+
+
+@compiles(sa.sql.elements.Extract, 'hive')
```

Comment:
impala has an extract function which is what the expression gets compiled to by default:

http://www.cloudera.com/content/cloudera/en/documentation/cloudera-impala/v2-0-x/topics/impala_datetime_functions.html?scroll=datetime_functions